app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -2,126 +2,97 @@ module ManageIQ::Providers
+    def collect_report_metrics(report_cols, start_time, end_time, interval)
+      filtered = @target.metrics_available.select { |metric| report_cols.nil? || report_cols.include?(metric['name']) }
+      collect_live_metrics(filtered, start_time, end_time, interval)
+    end
+
+    def process_stats(raw_stats)
       processed = Hash.new { |h, k| h[k] = {} }
-      %w(availability gauge counter_rate).each do |type|
-        next unless raw_stats.key?(type)
-        raw_data = raw_stats[type]
-        raw_data.each do |metric_id, buckets|
-          metric_name = metrics_ids_map[metric_id]
-          norm_data = sort_and_normalize(buckets)
-          norm_data.each do |bucket|
-            timestamp = Time.at(bucket['start'] / 1.in_milliseconds).utc.to_i
-            value = type == 'availability' ? bucket['uptimeRatio'] : bucket['avg']
-            processed.store_path(timestamp, metric_name, value)
-          end
+      raw_stats.each do |raw_data|
+        next unless raw_data['values']
+        raw_data['values'].each do |datapoint|
+          timestamp = datapoint.first
+          value = datapoint.last.to_f
+          processed.store_path(timestamp, raw_data['metric']['name'], value)
         end
       end
       processed
     end
 
-    def first_and_last_capture_for_metrics(metrics)
-      firsts, lasts = metrics.collect do |metric|
-        first_and_last_capture(metric)
-      end.transpose
-      [firsts, lasts]
-    end
-
-    def first_and_last_capture(metric)
-      validate_metric(metric)
-      case metric[:type]
-      when "GAUGE"        then min_max_timestamps(@gauges, metric[:id])
-      when "COUNTER"      then min_max_timestamps(@counters, metric[:id])
-      when "AVAILABILITY" then min_max_timestamps(@avail, metric[:id])
-      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
+    def first_and_last_capture(interval_name = "realtime")
+      now = Time.new.utc
+      one_week_before = now - ONE_WEEK
+      last = now
+      first = now
+      results = @prometheus_client.up_time(:feed_id => @target.feed,
+                                           :starts  => one_week_before.to_i,
+                                           :ends    => now.to_i,
+                                           :step    => '3600m')
+      if results.empty?
+        one_day_before = now - ONE_DAY
+        results = @prometheus_client.up_time(:feed_id => @target.feed,
+                                             :starts  => one_day_before.to_i,
+                                             :ends    => now.to_i,
+                                             :step    => '60m')
       end
-    end
-
-    def validate_metric(metric)
-      unless metric && %i(id name type unit).all? { |k| metric.key?(k) }
-        raise MetricValidationError, "Validation error: metric #{metric} must be defined"
+      unless results.empty?
+        datapoint = results.first
+        first = Time.at(datapoint.first.to_i)
```

Comment:
This line is outputting me a rubocop warning:
```
app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb:83 W: Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.

Reply:
Ok, I will review, but I guess that if it comes from a timestamp, it should be utc.